lib/charms/postgresql_k8s/v0/postgresql_tls.py

```diff
@@ -130,11 +130,21 @@ def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
             logger.error("An unknown certificate available.")
             return
 
-        self.charm.set_secret(
-            SCOPE, "chain", "\n".join(event.chain) if event.chain is not None else None
-        )
-        self.charm.set_secret(SCOPE, "cert", event.certificate)
-        self.charm.set_secret(SCOPE, "ca", event.ca)
+        if not event.certificate:
+            logger.debug("No certificate available.")
+            event.defer()
+            return
+
+        chain = self.charm.get_secret(SCOPE, "chain")
+        new_chain = "\n".join(event.chain) if event.chain is not None else None
+        if chain != new_chain:
+            self.charm.set_secret(SCOPE, "chain", new_chain)
+        cert = self.charm.get_secret(SCOPE, "cert")
+        if cert != event.certificate:
+            self.charm.set_secret(SCOPE, "cert", event.certificate)
+        ca = self.charm.get_secret(SCOPE, "ca")
+        if ca != event.ca:
+            self.charm.set_secret(SCOPE, "ca", event.ca)
```

Comment:
I think this fires a `secret_changed` event even if the secret value is the same.

Reply:
So, it's worth reporting it as either a DP-Libs or a Juju bug.